utils.py

```diff
@@ -2198,6 +2244,74 @@ def lcm_of_all_minors(G):
     return multi_polylcm(denoms)
 
 
+def poles_and_zeros_of_square_tf_matrix(G):
+    """
+    Determine poles and zeros of a square mimotf matrix, making use of the determinant.
+    This method may fail in special cases. If terms cancel out during calculation of the determinant,
+    not all poles and zeros will be determined.
+
+    Parameters
+    ----------
+    G : mimotf matrix (n x n)
+        The transfer function of the system.
+
+    Returns
+    -------
+    z : array
+        List of zeros.
+    p : array
+        List of poles.
+    cancel : boolean
+             Test whether any terms were cancelled out in determinant calculation.
+
+    Example
+    -------
+    >>> G = mimotf([[tf([1, -1], [1, 2]), tf([4], [1, 2])],
+    ...             [tf([4.5], [1, 2]), tf([2, -2], [1, 2])]])
+    >>> poles_and_zeros_of_square_tf_matrix(G)
+    (array([ 4.]), array([-2.]), False)
+    """
+
+    # Convert mimotf to sympy matrix
+    Gs, s = mimotf2sym(G, deadtime=False)
+
+    # Test cancellation
+    rows, cols = G.shape
+    r = Gs.rank()
+    cancel = False
+    for i in range(rows):
+        for j in range(cols):
+            minor_multiply = Gs[r*i + j]
+            minors = Gs.minor_submatrix(i, j)
+            for minor in minors:
+                res = minor_multiply//sympy.denom(minor)
+                if res == 0 and sympy.denom(minor) != 1:
+                    cancel = True
+
+    # Determine determinant
+    detG = Gs.det().simplify()
+
+    # Determine numerator and denominator
+    numer = sympy.numer(detG).expand()
+    denom = sympy.denom(detG).expand()
+
+    try:
+        numer_poly = sympy.Poly(numer, s)
+        denom_poly = sympy.Poly(denom, s)
+    except:
```

Comment:
You shouldn't use "naked except" and you shouldn't return bad values when an error occurs. Get rid of the try or raise a better exception with more information. And make sure your error is more informative than the one you're catching.